.github/workflows/rust.yml

```diff
@@ -55,3 +56,52 @@ jobs:
         run: |
           cargo build --all-features
           cargo test
+  windows-build:
+    runs-on: windows-latest
+    timeout-minutes: 30
+    defaults:
+      run:
+        working-directory: ./rust
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup cache
+        uses: actions/cache@v3
+        continue-on-error: false
+        with:
+          path: |
+            $HOME/.cargo/bin/
+            $HOME/.cargo/registry/index/
+            $HOME/.cargo/registry/cache/
+            $HOME/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Install Rustup using win.rustup.rs
```

Comment:
windows GHA has rust installed, can we use that one?

https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md